src/main/kotlin/no/nav/familie/ef/sak/vedtak/VedtakService.kt

```diff
@@ -64,14 +65,17 @@ class VedtakService(private val vedtakRepository: VedtakRepository) {
         return null
     }
 
-    fun hentForventetInntektForVedtakOgDato(behandlingIds: Collection<UUID>, dato: LocalDate): Map<UUID, Int?> {
+    fun hentForventetInntektForBehandlingIds(behandlingIds: Collection<UUID>): Map<UUID, Int?> {
         val vedtakList = vedtakRepository.findAllById(behandlingIds)
+        val dagensDatoMinusEnMåned = LocalDate.now().minusMonths(1)
         val map = mutableMapOf<UUID, Int?>()
         for (vedtak in vedtakList) {
-            if (vedtak?.perioder?.perioder?.any { it.datoFra.isEqualOrBefore(dato.minusMonths(1)) } == true) {
+            if (vedtak.erVedtakAktivtForDato(LocalDate.now())) {
```

Comment:
Hvis vedtak ikke lenger er aktivt så skal det heller ikke gis forventet inntekt

Reply:
Hva er det denne egentligen skal gjøre? 
Her hentes vedtaket, men siste vedtaket inneholder ju kanskje bare en revurdering frem i tid, og då ikke har inntekt fra forrige måned.

Burde vi bruke `TilkjentYtelse` ? Andelene i den inneholder inntekten for ønsket måned

Reply:
Det kan også være et opphør frem i tid - så tror vi må lene oss på tilkjentYtelse for at denne sjekken skal bli best mulig?

Reply:
Kan det løses ved å bruke en spørring?

```
SELECT b.id behandling_id, [noe mer snacks her?...]
        FROM gjeldende_iverksatte_behandlinger b
         JOIN fagsak f ON f.id = b.fagsak_id
        WHERE b.stonadstype=:stønadstype AND EXISTS(SELECT 1 FROM andel_tilkjent_ytelse aty
            JOIN tilkjent_ytelse ty ON aty.tilkjent_ytelse = ty.id
            WHERE ty.id = aty.tilkjent_ytelse 
            AND ty.behandling_id = b.id
          AND aty.stonad_tom >= :dato
          AND aty.belop > 0)
```

Disse HAR iallefall en stønad på det tidspunktet du er interessert i, men du vil kanskje ha litt mer data om ytelsen? 🤔

Reply:
Tenker jeg løser dette med et ekstra kall mot ef-sak fra personhendelse for å hente identer med aktive stønader. Tar i bruk gjeldende_iverksatte_behandlinger view